kbase-extension/static/kbase/js/widgets/function_output/kbaseReportView.js

```diff
@@ -126,397 +134,796 @@ define(
             self.objIdentity = self.buildObjectIdentity(this.options.workspace_name, this.options.report_name, null, this.options.report_ref);
 
             //self.objIdentity = {ref : "11699/2/77"};
-            self.ws.get_objects([self.objIdentity],
-                function (data) {
-                    self.reportData = data[0].data;
-                    self.render();
-                }, function (error) {
-                self.clientError(error);
+            
+            self.ws.get_objects([self.objIdentity])
+                .then(function (result) {
+                    self.reportData = result[0].data;
+                    return self.getLinks(self.reportData);
+                })
+                .then(function (links) {
+                    self.reportLinks = links;
+                    return self.render();
+                })
+                .catch(function (err) {
+                    self.clientError(err);
+                });
+
+            // self.ws.get_objects([self.objIdentity],
+            //     function (data) {
+            //         self.reportData = data[0].data;
+            //         self.render();
+            //     },
+            //     function (error) {
+            //         self.clientError(error);
+            //     });
+        },
+
+        wrapHtmlDoc: function (content) {
+            if (/<html/.test(content)) {
+                console.warn('Html document inserted into iframe');
+                return content;
+            }
+            var t = html.tag,
+                htmlTag = t('html'),
+                head = t('head'),
+                body = t('body');
+            return htmlTag([
+                head(),
+                body({
+                    style: {
+                        margin: '0px',
+                        padding: '0px',
+                        overflow: 'auto'
+                    }
+                }, [
+                    content
+                ])
+            ]);
+        },
+
+        makeIframe: function (arg) {
+            var t = html.tag,
+                div = t('div'),
+                script = t('script'),
+                iframe = t('iframe');
+
+            var iframeId = 'frame_' + html.genId(),
+                iframeOrigin = document.location.origin,
+                iframeMessages = HostMessages.makeHost({
+                    root: window,
+                    name: 'panel'
+                });
+
+            // The iframe content needs requirejs amd.
+
+            var requireConfig = {
+                    baseUrl: 'http://localhost:8888/narrative/static/',
```

Comment:
This should also be a configuration - it probably won't work anywhere but locally.